apps/backend/src/modules/routes/routes.repository.ts

```diff
@@ -0,0 +1,83 @@
+import {
+	type RoutesRequestPatchDto,
+	type RoutesResponseDto,
+} from "@smartscapes/shared";
+import { type QueryBuilder } from "objection";
+
+import { type Repository } from "~/libs/types/types.js";
+
+import { type RouteEntity } from "./routes.entity.js";
+import { type RoutesModel } from "./routes.model.js";
+
+class RoutesRepository implements Repository {
+	private routesModel: typeof RoutesModel;
+
+	public constructor(routesModel: typeof RoutesModel) {
+		this.routesModel = routesModel;
+	}
+
+	public async create(entity: RouteEntity): Promise<RoutesResponseDto> {
+		const insertData = entity.toNewObject();
+
+		const result = (await this.routesModel
+			.query()
+			.insertGraph(insertData, { relate: ["pois"] })
+			.returning([
+				"id",
+				"name",
+				"description",
+			])) as unknown as RoutesResponseDto;
+
+		return result;
```

Comment:
Check out how it's done in other places. We need to use entity initialize and return the entity from the repository level whenever we return an object. Also, we need to get rid of the typecast here; the entity should provide the necessary type. And the same for all other methods in the repository.

 Let me know if there are other issues with types here, and typecast is needed. Also, don't forget to follow naming conventions 

```suggestion
		const route = await this.routesModel
			.query()
			.insertGraph(insertData, { relate: ["pois"] })
			.returning([
				"id",
				"name",
				"description",
			]));

		return RoutesEntity.initialize(route);
```